src/handlers/debugHandler.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { default as getPort } from "get-port";
+import * as vscode from "vscode";
+import { createUuid } from "vscode-extension-telemetry-wrapper";
+import { PluginGoal } from "../explorer/model/PluginGoal";
+import { Settings } from "../Settings";
+import { executeInTerminal } from "../utils/mavenUtils";
+
+export async function debugHandler(goal: PluginGoal): Promise<void> {
+    // TODO: vscode-java-debug installation check
+    const pomPath: string = goal.plugin.project.pomPath;
+    const command: string = goal.name;
+    return await debug(pomPath, command);
+}
+
+async function debug(pomPath: string, command: string): Promise<void> {
+    const freePort: number = await getPort();
+    const mavenOpts: string = [
+        Settings.getEnvironment().MAVEN_OPTS, // user-setting MAVEN_OPTS
+        `-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=${freePort}` // MAVEN_DEBUG_OPTS
+    ].filter(Boolean).join(" ");
+    const sessionId: string = createUuid().substr(0, 6);
+    const debugTerimnal: vscode.Terminal = await executeInTerminal(command, pomPath, {
+        name: `mvnDebug ${sessionId}: ${command}`,
+        env: { MAVEN_OPTS: mavenOpts }

```

Comment:
Push the `process.env` into the `env` here as well?

Reply:
on creating a terminal, vscode already ensures that. 
```ts
		/**
		 * Object with environment variables that will be added to the VS Code process.
		 */
		env?: { [key: string]: string | null };
```